l10n_vn_account_invoice_number/models/account_invoice.py

```diff
@@ -0,0 +1,78 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 Trobz (http://trobz.com)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models, _
+from odoo.exceptions import UserError
+
+
+class AccountInvoice(models.Model):
+    _inherit = "account.invoice"
+
+    tax_invoice_number = fields.Integer()
+    tax_invoice_form = fields.Char()
+    tax_invoice_series = fields.Char()
+    is_issuer_in_vn = fields.Boolean(string='Is issuer in Vietnam',
+                                     store=False,
+                                     compute='_compute_is_issuer_in_vn')
+
+    @api.onchange('partner_id', 'company_id')
+    def _onchange_partner_id(self):
+        self._compute_is_issuer_in_vn()
+        return super(AccountInvoice, self)._onchange_partner_id()
+
+    @api.multi
+    def _compute_is_issuer_in_vn(self):
+        issuer_country_code = ''
+        if self.type == 'out_invoice' or self.type == 'in_refund':
+            issuer_country_code = self.company_id.partner_id.country_id.code
+        else:
+            issuer_country_code = self.partner_id.country_id.code
```

Comment:
Sometimes, the country won't be set and sometimes the country will be set only on the parent. Also, considering that this module is specific for VN, we can assume that if not set the partner is in vietnam:

So, I think we should have this code:
```
        if self.type == 'out_invoice' or self.type == 'in_refund':
            issuer_country_id = country_id and self.company_id.partner_id.country_id
        else:
            issuer_country_id = self.partner_id.country_id or self.partner_id.parent_id.country_id
        issuer_country_code = issuer_country_id and issuer_country_id.code or "VN"
```